tensorflow/compiler/xla/service/gpu/ir_emitter.cc

```diff
@@ -289,11 +291,42 @@ bool IrEmitter::MaybeEmitDirectAtomicOperation(
       }
     }
 
+    if (isAMDGPU) {
+      auto output_address_type =
```

Comment:
Due to nesting, this gets hard to read, maybe factor out the AMD emission to a separate function? Also `if (IsEmittingForAMDGPU())` seems more readable.

Reply:
done.